### PR TITLE
feat: add starred spaces

### PR DESCRIPTION
**Pull request by j0hnfl0w**
Closes: https://github.com/snapshot-labs/sx-ui/issues/383

- added @iconify-json/heroicons-solid for solid icons instead custom svg icon for star
- starred space stores in reactive localStorage with useStorage from vueuse, no login required
- added starred spaces under the fetched spaces in Sidebar, but we need only starred here?

### Screenshot
image

### CODE REVIEW

**Comment by j0hnfl0w** [comment]
Fixed - only starred spaces in Sidebar
image

File: src/stores/spaces.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { defineStore } from 'pinia';
 import { currentNetwork } from '@/networks';
+import { useStorage } from '@vueuse/core';
 import type { Space } from '@/types';
+import pkg from '@/../package.json';
```
Comment (Sekhmet) [inline src/stores/spaces.ts:5]:
```suggestion
import pkg from '../../package.json';
```

Going back an alias looks a bit ugly, let's use relative import for now.

File: src/stores/spaces.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { defineStore } from 'pinia';
 import { currentNetwork } from '@/networks';
+import { useStorage } from '@vueuse/core';
```
Comment (Sekhmet) [inline src/stores/spaces.ts:3]:
```suggestion
import { useStorage } from '@vueuse/core';
import { currentNetwork } from '@/networks';
```

nit. usually we put npm imports above internal ones.

File: src/stores/spaces.ts [hunk 3]
```diff
@@ -10,7 +12,8 @@ export const useSpacesStore = defineStore('spaces', {
     loadingMore: false,
     loaded: false,
     hasMoreSpaces: true,
-    spaces: [] as Space[]
+    spaces: [] as Space[],
+    spacesStarred: useStorage(`${pkg.name}.spaces-starred`, [] as string[])
```
Comment (Sekhmet) [inline src/stores/spaces.ts:16]:
```suggestion
    starredSpacesIds: useStorage(`${pkg.name}.spaces-starred`, [] as string[])
```
`starredSpacesIds` might be more easy to understand what is inside.

**Review by Sekhmet — COMMENTED** [review]
Other than it looks good to me.

@bonustrack what you might want to look into is that be default side bar will be empty until we add some space to favorites.

**Comment by j0hnfl0w** [comment]
Fixed all above and synced my fork repo with master

File: src/stores/spaces.ts [hunk 4]
```diff
@@ -10,7 +12,8 @@ export const useSpacesStore = defineStore('spaces', {
     loadingMore: false,
     loaded: false,
     hasMoreSpaces: true,
-    spaces: [] as Space[]
+    spaces: [] as Space[],
+    spacesStarredIds: useStorage(`${pkg.name}.spaces-starred`, [] as string[])
```
Comment (Sekhmet) [inline src/stores/spaces.ts:16]: nit. I imagine it's pretty nit-picky, but `spacesStarred` seems weird (compared to `starredSpaces`).

Comment (j0hnfl0w) [reply src/stores/spaces.ts]: Changed to ``starredSpacesIds``, but I prefer to start with 'spaces' prefix for all spaces related vars, its better readability but maybe not perfect grammatically or weird :)
```
const state = {
  ...another state vars
  spaces: [],
  spacesSubscribedIds: [],
  spacesStarredIds,
  spaceSelected: null,
  ...another state vars
}
```
vs
```
const state = {
  ...another state vars
  spaces: [],
  subscribedSpacesIds: [],
  starredSpacesIds: [],
  selectedSpace: []
  ...another state vars
}
```
What do you think?

Comment (j0hnfl0w) [reply src/stores/spaces.ts]: although there all vars related to spaces in spaces store )

**Comment by bonustrack** [comment]
Thanks @j0hnfl0w, we've merged another PR that changed the URLs for space to include a network, could you update your PR to fix conflicts and add the space network as prefix with the space address?

**Comment by j0hnfl0w** [comment]
Yep, will fix it today

File: src/components/Sidebar.vue [hunk 5]
```diff
@@ -12,16 +13,18 @@ onMounted(() => spacesStore.fetch());
...
+    <UiLoading v-if="!spacesStore.loaded" class="py-2 block" />
+    <draggable
+      v-else
+      v-model="spacesStore.starredSpacesIds"
+      :item-key="i => i"
+      class="space-y-3 p-2"
+    >
+      <template #item="{ element }">
+        <router-link :to="{ name: 'overview', params: { id: element } }" class="block">
+          <Stamp :id="element" :size="32" class="!rounded-[4px]" />
```
Comment (Sekhmet) [inline src/components/Sidebar.vue:25]: We should drop networkId for Stamp.

Comment (j0hnfl0w) [reply src/components/Sidebar.vue]: hm, but why these links:
[networkId+spaceId](https://stamp.fyi/avatar/gor:0x95DC6f73301356c9909921e21b735601C42fc1a8?s=64)
[spaceId](https://stamp.fyi/avatar/0x95DC6f73301356c9909921e21b735601C42fc1a8?s=64)
return the same image?

File: src/views/Space/Overview.vue [hunk 6]
```diff
@@ -8,16 +9,21 @@ const PROPOSALS_LIMIT = 4;
...
 });
 
-const proposalsRecord = computed(
-  () => proposalsStore.proposals[`${props.space.network}:${props.space.id}`]
-);
+const spaceIdComposite = `${props.space.network}:${props.space.id}`;
+
+const spaceStarred = computed(() => {
+  return spacesStore.starredSpacesIds.includes(spaceIdComposite);
+});
```
Comment (Sekhmet) [inline src/views/Space/Overview.vue:24]:
```suggestion
const spaceStarred = computed(() => spacesStore.starredSpacesIds.includes(spaceIdComposite));
```

nit.

**Review by Sekhmet — APPROVED** [review]
tACK

Looks good to me.

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**